### PR TITLE
refactor(provisioning_api): Replace security annotations with respective attributes

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Follow-up to https://github.com/nextcloud/server/pull/46606

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 1]
```diff
@@ -94,6 +93,8 @@ public function getGroups(string $search = '', ?int $limit = null, int $offset =
 	 *
 	 * 200: Groups details returned
 	 */
+	#[NoAdminRequired]
+	#[AuthorizedAdminSetting(Sharing::class)]
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/GroupsController.php:97]: The common pattern is to name the parameter 

```suggestion
	#[AuthorizedAdminSetting(setting: Sharing::class)]
```

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -783,6 +779,9 @@ public function getEditableFieldsForUser(string $userId): DataResponse {
 	 *
 	 * 200: User values edited successfully
 	 */
+	#[PasswordConfirmationRequired]
+	#[NoAdminRequired]
+	#[UserRateLimit(5, 60)]
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php:784]:
```suggestion
	#[UserRateLimit(limit: 5, period: 60)]
```

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 3]
```diff
@@ -885,6 +881,9 @@ public function editUserMultiValue(
 	 *
 	 * 200: User value edited successfully
 	 */
+	#[PasswordConfirmationRequired]
+	#[NoAdminRequired]
+	#[UserRateLimit(50, 60)]
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php:886]:
```suggestion
	#[UserRateLimit(limit: 50, period: 60)]
```

File: apps/provisioning_api/lib/Controller/VerificationController.php [hunk 4]
```diff
@@ -78,10 +81,10 @@ public function showVerifyMail(string $token, string $userId, string $key): Temp
 	}
 
 	/**
-	 * @NoAdminRequired
 	 * @NoSubAdminRequired
-	 * @BruteForceProtection(action=emailVerification)
 	 */
+	#[NoAdminRequired]
+	#[BruteForceProtection('emailVerification')]
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/VerificationController.php:87]:
```suggestion
	#[BruteForceProtection(action: 'emailVerification')]
```

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**